Updated dependencies to Angular2 Release version

feitzi opened the pull request:
I have updated all angular dependencies to the new release version.
I also changed the electron index.js to a typescript file

feitzi committed:
working version with angular2 release version

feitzi committed:
working version with angular2 release version

feitzi committed:
upgraded from deprecated tsd to typings

feitzi committed:
refactored electron index.ts to typescript code

feitzi committed:
removed css style from index.html

feitzi committed:
add orginal button and input field

c4wrd commented on src/systemjs.config.js:
```diff
@@ -1,52 +1,43 @@
+  System.config({
+    paths: {
+      // paths serve as alias
+      'npm:': '../node_modules/'
```
The issue with this is that the intention is to be able to build a distributable application, and doing this would require ALL node dependencies to be copied into the build. Maybe look at changing this to what it was before, and copying over dependencies into dist with Gulp?

feitzi committed:
better way to load vendor files for electron

feitzi commented:
I changed the way to load npm packages.

c4wrd committed:
Fixed RX dependency issue

There is an issue with using the carat, currently it causes 5.0.0-beta.1 to be installed, throwing an error. Removing the carat fixes it, enforcing the exact version needed by all of Angular 2 packages.

c4wrd reviewed (approved):
Thanks for your work, it's greatly appreciated!
